openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1044,7 +1081,7 @@ Cache-Control: no-store
 
 The Credential Issuer MAY provide the Client with a `c_nonce` as defined in (#credential-response) in a Credential Error Response using `invalid_proof` error code defined in (#credential-error-response) if the Credential Issuer Metadata contains `proof_types_supported` indicating a key proof is required for the requested Credential. Depending on the Credential Issuer policy, this occurs if they receive a Credential Request without a `c_nonce` or with an invalid `c_nonce` value included in the proof(s) in the `proof` or `proofs` parameter.
 
-If the Client has not received a `c_nonce` and the Credential Issuer Metadata contains `proof_types_supported` indicating a key proof is required for the requested Credential, the Client MUST send a Credential Request that contains a `proof` or `proofs` parameter that is fully valid but does not include a `c_nonce` value. It is the Credential Issuer policy whether or not a `c_nonce` value is required in the key proofs.
+If the Client has not received a `c_nonce` and the Credential Issuer Metadata contains `proof_types_supported` indicating a key proof is required for the requested Credential, the Client can send a Credential Request that contains a `proof` or `proofs` parameter that is fully valid but does not include a `c_nonce` value. It is the Credential Issuer policy whether or not a `c_nonce` value is required in the key proofs.
```

Comment:
When is this paragraph apply? I'm asking since the client can always obtain a c_nonce from the nonce endpoint.

Reply:
+1, I was trying to push in this direction as well: https://github.com/openid/OpenID4VCI/pull/381#issuecomment-2366932177

Reply:
As I've tried to convey in comments on this PR and in calls over the course of the last month, I attempted to keep the scope of this change narrow so as to hopefully make it acceptable to the wide range of opinions and oftentimes irregular participation of the WG members. I have clearly failed in my efforts. I believe the text in question here was written recently by @awoie and because it is "Credential Issuer policy whether or not a `c_nonce` value is required in the key proofs" and a "Client can send a Credential Request that contains a `proof` or `proofs` parameter that is fully valid but does not include a `c_nonce` value" it seemed like it applied enough to not delete it.

Reply:
One reason why I requested the addition of the nonce endpoint was to avoid this kind of incomplete request just to trigger the provision of a fresh nonce. The wallet know that it needs a nonce, so it can directly request it from the nonce endpoint. 
I would suggest to delete this paragraph. If needed, similar text should be added  

```
If the Client is not in possession of a valid `c_nonce` and the Credential Issuer Metadata contains `proof_types_supported` indicating a key proof is required for the requested Credential, the Client MUST obtain a fresh nonce from the nonce endpoint. 
```

Reply:
So how does the wallet know it needs to include a proof?

Reply:
I think that the current text says that the presence of `proof_types_supported` in the metadata indicates that the prof is required in the request

https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#name-credential-issuer-provided-

Reply:
I think I now understand @tlodderstedt's point.

Some of this overlaps with https://github.com/openid/OpenID4VCI/issues/393 but not completely.

How do people feel about this making the use of the nonce_endpoint by the client mandatory if it's present and a nonce is required:

```suggestion
If the Credential Issuer Metadata contains a `nonce_endpoint` and a `proof_types_supported` indicating a key proof is required for the requested Credential, the Client MUST obtain a `c_nonce` value from the `nonce_endpoint` and send a Credential Request that contains a `proof` or `proofs` parameter that includes a `c_nonce` value. It is the Credential Issuer policy whether or not a `c_nonce` value is required in the key proofs.
```

Reply:
If I get it right, the issuer would signal the requirement to add the nonce to the proof by the existence of its nonce endpoint. 
That's ok with me.

Reply:
Correct, that is my suggestion. If a issuer requires a nonce for the proofs for some credentials and not others (not sure if this is a likely case) then this may end up with the client provided a nonce unnecessarily, but that doesn't really hurt anything.

Reply:
I support this text. It clarifies how the c_nonce should work

Reply:
the situation was supposed to be accounted for by the text "A Credential Issuer that requires `c_nonce` values to be incorporated into proofs in the Credential Request (see (#credential-request)) MUST offer a Nonce Endpoint." above in the Nonce Endpoint {#nonce-endpoint} section while not disturbing the existing optionality and whatever on nonce in the existing text with the nonce-by-error (or success) but I don't know anymore and apparently that was a dumb thing to do

Reply:
if it'll help move this forward, i'll commit the suggestion